conanfile.py

```diff
@@ -1,112 +1,81 @@
 from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
 import os
-import shutil
-import re
+
 
 class MongoCxxConan(ConanFile):
     name = "mongo-cxx-driver"
-    version = "3.3.0"
+    version = "3.4.0"
     url = "http://github.com/bincrafters/conan-mongo-cxx-driver"
     description = "C++ Driver for MongoDB"
-    license = "https://github.com/mongodb/mongo-cxx-driver/blob/{0}/LICENSE".format(version)
-    settings =  "os", "compiler", "arch", "build_type"
-    options = {"shared": [True, False]}
-    default_options = {'shared': 'False'}
-    requires = 'mongo-c-driver/[~=1.9]@bincrafters/stable'
+    license = "Apache-2.0"
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "polyfill": ["std", "boost", "mnmlstc", "experimental"]
+    }
+    default_options = {"shared": False, "fPIC": True, "polyfill": "boost"}
+    requires = "mongo-c-driver/1.16.1@bincrafters/stable"
     generators = "cmake"
 
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.compiler == 'Visual Studio' and self.options.polyfill != "boost":
+            raise ConanInvalidConfiguration("For MSVC, best to use the boost polyfill")
+
+        if self.options.polyfill == "std":
+            tools.check_min_cppstd(self, "17")
+
+        if self.options.polyfill == "boost":
```

Comment:
You can't verify options in config_options, any option won't be evaluated here. Move it to configure.